components/layout/SideNav.tsx

```diff
@@ -186,17 +219,48 @@ const GroupToggle = styled.div`
   justify-content: space-between;
   align-items: center;
   cursor: pointer;
-
-  > .caret {
-    margin-right: 10px;
-  }
 `;
 
-const StyledIcon = styled.img`
-  width: 17px;
-  height: 17px;
+const GroupCaret = styled.img`
+  max-width: 7px;
 `;
 
+type DocGroupProps = {
+  title: string;
+  docs: Page[];
+};
+
+const DocGroup = ({ title, docs }: DocGroupProps) => {
+  const { pathname } = useRouter();
+  const [isExpanded, setIsExpanded] = useState(
+    !!docs.find((d) => d.id === pathname)
+  );
+  useEffect(() => {
+    setIsExpanded(docs.find((d) => d.id === pathname) ? true : isExpanded);
+  }, [pathname]);
```

Comment:
Tracking whether a group is expanded at the group level rather than the side-nav level. This made it significantly easier to make sure the group containing the active doc is expanded when accessing from an index page or on refresh